### PR TITLE
docs/release-cycle.md: update the document

### PR DESCRIPTION
**Pull request by neolit123**
/assign @fabriziopandini 
/kind documentation
/priority backlog

fixes https://github.com/kubernetes/kubeadm/issues/1766

- tried to use the original doc but iterate on some of the aspects that have changed.
- amended the indentation so the diff should be smaller otherwise.

### CODE REVIEW

File: docs/release-cycle.md [hunk 1]
```diff
@@ -17,65 +16,62 @@ That said, nothing here is set in stone. See this as inspiration :)
...
+  - Make sure high-priority bug reports are triaged and fixed.
+  - Create bug fixes, ensure they're merged, and cherry pick them to the release branch.
+  - Coordinate with the patch release team.
+- Plan/prioritize features, testing work and bugs
+  - At the beginning of each cycle the kubeadm team has a planning session during the kubeadm office hours meeting.
+  - There people sign up for working on the high level features and objectives for the release.
+  - Based on the planning session shift issues between milestones in the `kubernetes/kubeadm` repository.
+  Create new issues if needed.
+- Go through _every_ open issue in the kubeadm repository.
+  - As there are a small number of core maintainers working on kubeadm, there often isn't time to answer
```
Comment (fabriziopandini) [inline docs/release-cycle.md]: I think we should move this phrase after “As a general practice...." and simplify into
- close old, outdated issues.

File: docs/release-cycle.md [hunk 2]
```diff
@@ -17,65 +16,62 @@ That said, nothing here is set in stone. See this as inspiration :)
...
+- Remove e2e tests for k8s/kubeadm versions outside of the support skew.
+- Manage cherrypicks for the current stable branch
+  - Make sure high-priority bug reports are triaged and fixed.
+  - Create bug fixes, ensure they're merged, and cherry pick them to the release branch.
+  - Coordinate with the patch release team.
+- Plan/prioritize features, testing work and bugs
+  - At the beginning of each cycle the kubeadm team has a planning session during the kubeadm office hours meeting.
+  - There people sign up for working on the high level features and objectives for the release.
+  - Based on the planning session shift issues between milestones in the `kubernetes/kubeadm` repository.
+  Create new issues if needed.
```
Comment (fabriziopandini) [inline docs/release-cycle.md]: IMO the issue part goes after the KEP part; might be worth to add something on the lines of.
- break down the implementation of larger feature into smaller actionable issue, so the work can be federated
- when possible, create help want/good for first contributor issues
- use a tracking issue to track/coordinate all the efforts

File: docs/release-cycle.md [hunk 3]
```diff
@@ -17,65 +16,62 @@ That said, nothing here is set in stone. See this as inspiration :)
...
- - Until pushing of debs/rpms is an automated process, ping a Googler to push newly generated debs and rpms to the respective repos, can be verified via these links:
-   - debs: https://packages.cloud.google.com/apt/dists/kubernetes-xenial/main/binary-amd64/Packages
-   - rpms: https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64/repodata/primary.xml
+- This month; code freeze is applied and no feature PRs may merge, only critical bug fixes
+  - Right before code freeze you should ensure that all feature PRs for kubeadm are reviewed, LGTM'd and approved.
+- Attend the Release Burndown Meetings whenever possible and/or communicate with the release team often to be on top of the release events.
+- Marketing & announcements (optional)
+  - Write a blog post on new features, possibly for inclusion in the "Five days of Kubernetes v1.X" blog post series on https://blog.kubernetes.io
+  - Demonstrate new features at the Kubernetes Community Meeting.
+  - Sync with Developer Advocates and collaborate on writing about kubeadm features, also those who aren't the most highlighted one, but still useful
```
Comment (fabriziopandini) [inline docs/release-cycle.md]: Is it possible/advisable to document how to contact dev advocates?

Comment (neolit123) [reply docs/release-cycle.md]: i don't know how to contact them.

Comment (neolit123) [reply docs/release-cycle.md]: deleted this sentence.

**Comment by neolit123** [comment]
@fabriziopandini updated. PTAL.

**Pull request merged**